booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,645 @@
+  }
+  private static final JsonContentCheck IS_EXPORT_DECLARATION_REFERENCE_PRESENT = JsonAttribute.ifThen(
+    "Check Export declaration reference ",
+    JsonAttribute.isTrue(JsonPointer.compile("/isExportDeclarationRequired")),
+    JsonAttribute.mustBePresent(JsonPointer.compile("/exportDeclarationReference"))
+  );
+
+  private static final JsonContentCheck IS_IMPORT_DECLARATION_REFERENCE_PRESENT = JsonAttribute.ifThen(
+    "Check Import declaration reference presence",
+    JsonAttribute.isTrue(JsonPointer.compile("/isImportLicenseRequired")),
+    JsonAttribute.mustBePresent(JsonPointer.compile("/importLicenseReference"))
+  );
+
+  private static final JsonContentCheck DOCUMENT_PARTY_FUNCTIONS_MUST_BE_UNIQUE = JsonAttribute.customValidator(
+    "Each document party can be used at most once",
+    JsonAttribute.path(
+      "documentParties",
+      JsonAttribute.unique("partyFunction")
+    ));
+
+  private static final JsonContentCheck COMMODITIES_SUBREFERENCE_UNIQUE = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Each Subreference in commodities must be unique",
+    (mav) -> {
+      mav.submitAllMatching("requestedEquipments.*.commodities");
+    },
+    JsonAttribute.unique("subReference")
+  );
+
+  private static final JsonContentCheck VALIDATE_ALLOWED_SHIPMENT_CUTOFF_CODE = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Validate allowed shipment cutoff codes",
+    (mav) -> mav.submitAllMatching("shipmentCutOffTimes.*.cutOffDateTimeCode"),
+    JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.CUTOFF_DATE_TIME_CODES)
+  );
+
+  private static final JsonContentCheck VALIDATE_SHIPMENT_CUTOFF_TIME_CODE = JsonAttribute.customValidator(
+    "validate shipment cutOff Date time code",
+    (body) -> {
+      var shipmentCutOffTimes = body.path("shipmentCutOffTimes");
+      var receiptTypeAtOrigin = body.path("receiptTypeAtOrigin").asText("");
+      var issues = new LinkedHashSet<String>();
+      var cutOffDateTimeCodes = StreamSupport.stream(shipmentCutOffTimes.spliterator(), false)
+        .map(p -> p.path("cutOffDateTimeCode"))
+        .filter(JsonNode::isTextual)
+        .map(n -> n.asText(""))
+        .collect(Collectors.toSet());
+      if (receiptTypeAtOrigin.equals("CFS") && !cutOffDateTimeCodes.contains("LCL")) {
+        issues.add("cutOffDateTimeCode 'LCL' must be present when receiptTypeAtOrigin is CFS");
+      }
+      return issues;
+    }
+  );
+
+  private static final Consumer<MultiAttributeValidator> ALL_AMF = (mav) -> mav.submitAllMatching("advanceManifestFilings.*");
+
+  private static final JsonContentCheck ADVANCED_MANIFEST_FILING_CODES_UNIQUE = JsonAttribute.allIndividualMatchesMustBeValid(
+    "The combination of 'countryCode' and 'manifestTypeCode' in 'advanceManifestFilings' must be unique",
+    ALL_AMF,
+    JsonAttribute.unique("countryCode", "manifestTypeCode")
+  );
```

Comment:
Have you validated this one (as in deliberately try to make it fail to ensure it flags invalid input)? unique only works when used on an (unexpanded) array, but the validator seems to validate the array.

Reply:
The `unique` condition requires the same element / key twice or more before it can trigger.  It does not sound like it was properly tested. Please try with a repeated version and report back.